File: components/physics/src/main/java/org/storm/physics/entity/ImmovableEntity.java
package org.storm.physics.entity;

import lombok.NonNull;
import org.storm.physics.math.Vector;
import org.storm.physics.math.geometry.shapes.Shape;

/**
 * An ImmovableEntity is a normal Entity but cannot be moved.
 */
public class ImmovableEntity extends Entity {

  public ImmovableEntity(@NonNull Shape hurtBox) {
    super(hurtBox, 0, Double.POSITIVE_INFINITY, 1);
  }

  @Override
  public void translate(Vector vector) {
    // ImmovableEntity does not move
  }

  @Override
  public void translate(double dx, double dy) {
    // ImmovableEntity does not move
  }

}

File: components/physics/src/main/kotlin/org/storm/physics/entity/ImmovableEntity.kt
package org.storm.physics.entity

import org.storm.physics.math.Vector
import org.storm.physics.math.geometry.shapes.Shape

/**
 * An ImmovableEntity is a normal Entity but cannot be moved.
 */
open class ImmovableEntity(boundaries: MutableMap<String, Shape>) : Entity(boundaries, 0.0, Double.POSITIVE_INFINITY, 1.0) {

    constructor(boundary: Shape) : this(mutableMapOf(SINGLE_BOUNDARY to boundary))

    override fun translate(dx: Double, dy: Double) {
        // ImmovableEntity does not move
    }

}
